docs/ref/forms/api.txt

```diff
@@ -1441,21 +1464,17 @@ Methods of ``BoundField``
         >>> print(bound_form["subject"].value())
         hi
 
+.. _custom-boundfield:
+
 Customizing ``BoundField``
 ==========================
 
-If you need to access some additional information about a form field in a
-template and using a subclass of :class:`~django.forms.Field` isn't
-sufficient, consider also customizing :class:`~django.forms.BoundField`.
-
-A custom form field can override ``get_bound_field()``:
-
-.. method:: Field.get_bound_field(form, field_name)
+A form can override :attr:`.Form.bound_field_class` and a form field can take
+an optional :attr:`.Field.bound_field_class` constructor argument.
 
-    Takes an instance of :class:`~django.forms.Form` and the name of the field.
-    The return value will be used when accessing the field in a template. Most
-    likely it will be an instance of a subclass of
-    :class:`~django.forms.BoundField`.
+For example, you may need to access some additional information about a form
+field in a template and using a subclass of :class:`~django.forms.Field` isn't
+sufficient.
 
 If you have a ``GPSCoordinatesField``, for example, and want to be able to
 access additional information about the coordinates in a template, this could
```

Comment:
I'm sorry I didn't realize that `get_bound_field` is documented in an old release notes and overriding it is also covered in tests, given that I think this should start with:
```
There are multiple ways to use a custom :class:`.BoundField`:

* A project can define a project-wise :class:`.BoundField` class using
  :attr:`.BaseRenderer.bound_field_class` and :setting:`FORM_RENDERER`.

* A form can override :attr:`.Form.bound_field_class` or use the constructor
  argument.

* A custom form field can override :attr:`.Field.bound_field_class` or use the
  constructor argument.

* A custom form field can override :meth:`.Field.get_bound_field()`:

.. method:: Field.get_bound_field(form, field_name)

    Takes an instance of :class:`~django.forms.Form` and the name of the field.
    The return value will be used when accessing the field in a template. Most
    likely it will be an instance of a subclass of :class:`.BoundField`.

You may want to use a custom :class:`.BoundField` if you need to access some
additional information about a form field in a template and using a subclass of
:class:`~django.forms.Field` isn't sufficient.

For example, if you have a ``GPSCoordinatesField``, for example, and want to be
able to access additional information about the coordinates in a template, this
could be implemented as follows::
```
(sorry for the back and forth)